client/fingerprint_manager.go

```diff
@@ -0,0 +1,126 @@
+package client
+
+import (
+	"log"
+	"time"
+
+	"github.com/hashicorp/nomad/client/config"
+	"github.com/hashicorp/nomad/client/driver"
+	"github.com/hashicorp/nomad/client/fingerprint"
+	cstructs "github.com/hashicorp/nomad/client/structs"
+	"github.com/hashicorp/nomad/nomad/structs"
+)
+
+// FingerprintManager runs a client fingerprinters on a continuous basis, and
+// updates the client when the node has changed
+type FingerprintManager struct {
+	getConfig  func() *config.Config
+	node       *structs.Node
+	shutdownCh chan struct{}
+
+	// updateNode is a callback to the client to update the state of its
+	// associated node
+	updateNode func(*cstructs.FingerprintResponse)
```

Comment:
Should this maybe return the updated node back? All the fingerprinters will be operating on stale nodes right now

Reply:
Hm, I was under the impression that the fingerprint manager just holds a pointer to the original node, which is only updated (not replaced). If this isn't true (or not the pattern that we want) then it can re-assign the node based on what this function returns.